tpm2_pytss/types.py

```diff
@@ -732,399 +743,715 @@ class TPMA_MEMORY(int):
     OBJECTCOPIEDTORAM = lib.TPMA_MEMORY_OBJECTCOPIEDTORAM
 
 
-### handy contructors
+class TPM_OBJECT(object):
+    def __init__(self, cdata):
+        self._cdata = cdata
+
+    @staticmethod
+    def _fixup_classname(tipe):
+        # Some versions of tpm2-tss had anonymous structs, so the kind will be struct
+        # but the name will not contain it
+        if tipe.cname.startswith(tipe.kind):
+            return tipe.cname[len(tipe.kind) + 1 :]
+
+        return tipe.cname
+
+    @staticmethod
+    def _init_cdata(child, cdata):
+
+        init_data = cdata
+        if cdata is None:
+            init_data = ffi.new(f"{child.__class__.__name__} *")
```

Comment:
> I'm not entirely clear on why this isn't just a part of the base class's `__init__`, or called within it. `self.__class__.__name__` would be the name of the subclass if this was a regular method and called from the base class `__init__`. 
> Seems like that would eliminate the repitiion of
> ```python
>     def __init__(self, _cdata=None):
>         init_data = TPM_OBJECT._init_cdata(self, _cdata)
>         super().__init__(init_data)
> ```

Reply:
My reason was I didn't think this worked as expected:
```python
class FOO(object):
    def __init__(self):
        print(self.__class__.__name__)

class BAR(FOO):
    def __init__(self):
        super().__init__()

f = FOO()
b = BAR()
```
```bash
python3 ~/a.py
FOO
BAR
```
But it does, so I can get rid of all these and just dynamically create them.